truffle/src/main/java/org/jruby/truffle/nodes/core/ProcessNodes.java

```diff
@@ -84,6 +84,7 @@ protected Object clock_gettime_realtime(int clock_id, RubyBasicObject unit) {
             return timeToUnit(time, unit);
         }
 
+        @TruffleBoundary
```

Comment:
This one might actually be specific to ahead of time compilation, what causes the explosion is when the jnr code makes something reachable that is unsupported, calling error handling code. I'll look into this a little more.